### PR TITLE
velero: add darwin as supported platform

### PR DESCRIPTION
**Pull request by rguevara84**
###### Motivation for this change
Velero supports Linux and darwin

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by qaifshaikh — APPROVED** [review]

File: pkgs/applications/networking/cluster/velero/default.nix [hunk 1]
```diff
@@ -40,6 +40,6 @@ buildGoModule rec {
       "https://github.com/vmware-tanzu/velero/releases/tag/v${version}";
     license = licenses.asl20;
     maintainers = [ maintainers.mbode maintainers.bryanasdev000 ];
-    platforms = platforms.linux;
+    platforms = [ "x86_64-linux" "x86_64-darwin" ];
```
Comment (kalbasit) [inline pkgs/applications/networking/cluster/velero/default.nix:43]:
```suggestion
    platforms = platforms.linux ++ platforms.darwin;
```

**Review by kalbasit — CHANGES_REQUESTED** [review]

**Review by qaifshaikh — APPROVED** [review]

**Review by bryanasdev000 — APPROVED** [review]
LGTM.

**Review by bryanasdev000 — APPROVED** [review]

**Review by kalbasit — APPROVED** [review]
Result of `nixpkgs-review pr 117278` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>velero</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/velero/default.nix [hunk 2]
```diff
@@ -40,6 +40,6 @@ buildGoModule rec {
       "https://github.com/vmware-tanzu/velero/releases/tag/v${version}";
     license = licenses.asl20;
     maintainers = [ maintainers.mbode maintainers.bryanasdev000 ];
-    platforms = platforms.linux;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/velero/default.nix:43]: This should be platforms.unix.

Comment (bryanasdev000) [reply pkgs/applications/networking/cluster/velero/default.nix:43]: @SuperSandro2000 I will do a PR to refactor these things in my packages, it was my mistake.

**Pull request merged**